### PR TITLE
make onClick optional for buttons in case they are used as submit butttons in forms

### PR DESCRIPTION
**Pull request by MagdalenaHenke**
[Clubhouse](https://app.clubhouse.io/blocks/story/34492/buttons-require-onclick-but-form-buttons-might-not-have-that)

Clubhouse describes how buttons in forms might not have an `onClick` function, so the `onClick` should be optional. Probably rarely the case, but still.

I tested this by throwing this into `ButtonBasicExample`:

```jsx
import React from "react"

const { ButtonPrimary, ButtonSecondary, ButtonDanger } = require('blocks-react').Buttons;

const ButtonBasicExample = () => (
  <form onSubmit={(e) => {console.log('Submitted!'); e.preventDefault()}}>
    <input type="text" name="firstname" />
    <ButtonPrimary
      text="Primary Buttons"
      type="submit"
      onClick={() => console.log('Primary Button Clicked')}
    />
  </form>
);

export default ButtonBasicExample;
```

### CODE REVIEW

**Comment by MagdalenaHenke** [comment]
https://app.clubhouse.io/blocks/story/34492/buttons-require-onclick-but-form-buttons-might-not-have-that

**Comment by MagdalenaHenke** [comment]
@Jinsoo94 @emma-floored This one's ready for review.

File: packages/blocks-react/buttons/Button.jsx [hunk 1]
```diff
@@ -29,26 +30,29 @@ const BasicButton = React.forwardRef((props, ref) => {
...
+  onClick: PropTypes.func,
   text: PropTypes.string,
   isDisabled: PropTypes.bool,
   leftIcon: PropTypes.node,
   rightIcon: PropTypes.node,
   icon: PropTypes.node,
   autoFocus: PropTypes.bool,
   onKeyDown: PropTypes.func,
-  onKeyUp: PropTypes.func
+  onKeyUp: PropTypes.func,
```
Comment (Jinsoo94) [inline packages/blocks-react/buttons/Button.jsx]: remove comma?

**Comment by Jinsoo94** [comment]
👍

**Comment by emma-floored** [comment]
👍 

also, this has made me wonder about creating separate uncontrolled components for forms (ex. checkboxes and radio buttons)

**Pull request merged**